client/chain/chain.go

```diff
@@ -634,8 +648,8 @@ type DerivativeOrderData struct {
 
 type SpotOrderData struct {
 	OrderType    exchangetypes.OrderType
-	Price        cosmtypes.Dec
-	Quantity     cosmtypes.Dec
```

Comment:
We should use `cosmtypes.Dec`everywhere instead of `decimal.Decimal` from `github.com/shopspring/decimal` package.

The chain uses cosmtypes so using it consistently on both client & server may save us some times debugging some weird conversion, format issues.

Reply:
I use decimal.Decimal in these structs to make the conversions for tick sizes on quantity/price, the functions that return the values that will be sent to the chain are always cosmtypes.Dec.

See: https://github.com/InjectiveLabs/sdk-go/blob/f/load_metadata_examples/client/chain/chain.go#L534-L581